### DISCUSSION TITLE
Setting Up on WSL

### DISCUSSION THREAD

**Discussion by jasoon-chen** [empriselab/RCareWorld#106] [Help - Setting up / Configuring] [answered]
I've been trying to setup the PyCareWorld environment on windows 11 using WSL 2 using Ubuntu 20.04.6, however when running `python test_scene.py` it is always stuck on `Waiting for connections on port: 5005...` in the terminal, but no windows pop up or anything.

I tried allowing port 5005 through inbound and outbound rules, but it still doesn't seem to load Unity up. I am also on the `phy-robo-care` branch.

Do you have any ideas what could be the issue here?

**Comment by shxingch** [comment]
Hi @jasoon-chen !
Please try to modify the `teat_scene.py` file by adding the `port=5006` statement to the code. Change:

```python
env = RCareWorld(assets=["Collider_Box", "Rigidbody_Sphere"], executable_file=player_path)
```

to:

```python
env = RCareWorld(assets=["Collider_Box", "Rigidbody_Sphere"], executable_file=player_path, port=5006)
```

**Reply by jasoon-chen** [reply]
I tried adding the `port=5006`, but it seems to now just be stuck on `Waiting for connections on port: 5007...`

When running the script and running `netstat -an | findstr 5007` on terminal it does mention that port 5007 is listening so something is going on. `TCP    127.0.0.1:5007         0.0.0.0:0              LISTENING`. I also ran the samething on port 5006 but it appears that nothing is being listened on port 5006.

**Reply by shxingch** [reply, marked as answer]
It doesn't seem to be a port issue then. Do you have graphical settings installed on your WSL? By default, it cannot display graphical windows and requires additional installation.
